shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -356,6 +361,46 @@ - (BOOL)launchEngineInternalWithAssetsPath:(NSURL*)assets
   flutterArguments.command_line_argv = argv;
   flutterArguments.platform_message_callback = (FlutterPlatformMessageCallback)OnPlatformMessage;
 
+  // Provide data for release mode engine.
+#if NDEBUG
+  const char* vmSnapshotDataPath = [[[NSBundle mainBundle] pathForResource:kVmSnapshotDataPath
+                                                                    ofType:nil] UTF8String];
+  const char* vmSnapshotInstructionsPath =
+      [[[NSBundle mainBundle] pathForResource:kVmSnapshotInstructionsPath ofType:nil] UTF8String];
+  const char* isolateSnapshotDataPath =
+      [[[NSBundle mainBundle] pathForResource:kIsolateSnapshotDataPath ofType:nil] UTF8String];
+  const char* isolateSnapshotInstructionsPath =
+      [[[NSBundle mainBundle] pathForResource:kIsolateSnapshotInstructionsPath
+                                       ofType:nil] UTF8String];
+  size_t vmSnapshotDataLength;
+  void* vmSnapshotData;
+  if (MapFile(vmSnapshotDataPath, &vmSnapshotData, &vmSnapshotDataLength, false) == 0) {
+    flutterArguments.vm_snapshot_data = (const uint8_t*)vmSnapshotData;
+    flutterArguments.vm_snapshot_data_size = vmSnapshotDataLength;
+  }
+  size_t vmSnapshotInstructionsLength;
+  void* vmSnapshotInstructions;
+  if (MapFile(vmSnapshotInstructionsPath, &vmSnapshotInstructions, &vmSnapshotInstructionsLength,
+              true) == 0) {
+    flutterArguments.vm_snapshot_instructions = (const uint8_t*)vmSnapshotInstructions;
```

Comment:
Why not pass the struct fields in directly, to avoid all the intermediate variables?

Reply:
I don't know C/C++ so I'm really just guessing here :)